flexmeasures/data/models/planning/storage.py

```diff
@@ -1030,6 +1034,28 @@ def compute(self, skip_validation: bool = False) -> SchedulerOutputType:
             for sensor in sensors
         }
 
+        flex_model = self.flex_model
+
+        if not isinstance(self.flex_model, list):
+            flex_model["sensor"] = sensors[0]
+            flex_model = [flex_model]
+
+        soc_schedule = {
+            flex_model_d["state_of_charge"]: integrate_time_series(
+                series=ems_schedule[d],
+                initial_stock=soc_at_start[d],
+                up_efficiency=device_constraints[d]["derivative up efficiency"].fillna(
+                    1
+                ),
+                down_efficiency=device_constraints[d][
+                    "derivative down efficiency"
+                ].fillna(1),
+                storage_efficiency=device_constraints[d]["efficiency"].fillna(1),
+            )
+            for d, flex_model_d in enumerate(flex_model)
+            if isinstance(flex_model_d.get("state_of_charge", None), Sensor)
+        }
+
         # Resample each device schedule to the resolution of the device's power sensor
```

Comment:
Let's make sure the soc sensor is instantaneous. Better yet, let's demand that in the schema's field validator.